### ISSUE TITLE
cloud_functions useFunctionsEmulator doesn't work

### ISSUE THREAD

**Issue by dmvvilela** [FirebaseExtended/flutterfire#3505]
I tried everything, the best i could do was this:

```dart
 CloudFunctions.instance.useFunctionsEmulator(
    origin: 'http://localhost:5001',
  );
```

and it gave me this error when i call a callable function:

```
The resource could not be loaded because the App Transport Security policy requires the use of a secure connection
```

But if i use https i does not work either.

I tried my ip, i tried localhost only - without http (in which btw firestore works ok).

Am i doing something wrong or is it a bug?

Also.. shouldn't the names be normalized like Firestore is now FirebaseFirestore.. maybe then it should be FirebaseFunctions with the same API?

**Comment by TahaTesser** [comment]
Hi @dmvvilela 
Please check out  https://flutter-firebase-docs.web.app/docs/firestore/usage#emulator-usage
Let me if this helps 
Thank you

**Comment by dmvvilela** [comment]
Hey,

For firestore it works. For cloud functions no.

I'm using iOS. localhost:port does not work.

**Comment by dmvvilela** [comment]
Here is my flutter doctor if anyone needs it:

```shell
~ % flutter doctor -v
[✓] Flutter (Channel stable, 1.20.3, on Mac OS X 10.15.6 19G2021, locale pt-BR)
    • Flutter version 1.20.3 at /Users/danvilela/flutter
    • Framework revision 216dee60c0 (10 days ago), 2020-09-01 12:24:47 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/danvilela/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (1 available)
    • iPhone 8 (mobile) • 289DE10C-FB13-45C6-B0CE-1392E20C0AFF • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-7 (simulator)

• No issues found!
```

**Comment by TahaTesser** [comment]
Related https://github.com/FirebaseExtended/flutterfire/issues/2792

**Comment by ranjithrd** [comment]
I faced the same error. I solved it using these steps.

I hope this helps someone. I'm willing to make a PR for this and add it under the emulator doc.

1. Open `ios/Runner.xcworkspace` in Xcode.
2. Open `Runner/info.plist`.
3. Add a property called `App Transport Security Settings`.
4. Right click on that property and click `Add Row`.
5. Add a property called `Allows Local Networking` and enter its value as `YES`.
